lib/screens/tabs_screen.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:flutter/material.dart';
+import 'package:meals_app/screens/categories_screen.dart';
+import 'package:meals_app/screens/favorites_screen.dart';
+
+class TabsScreen extends StatefulWidget {
+  @override
+  _TabsScreenState createState() => _TabsScreenState();
+}
+
+class _TabsScreenState extends State<TabsScreen> {
+  final List<Map<String, Object>> _pages = [
+    {
+      'page': CategoriesScreen(),
+      'title': 'Categories',
+    },
+    {
+      'page': FavoritesScreen(),
+      'title': 'Your favorite',
+    }
+  ];
+
+  int _selectedPageIndex = 0;
+
+  void _selectPage(int index) {
+    setState(() {
+      _selectedPageIndex = index;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(_pages[_selectedPageIndex]['title']),
+      ),
+      body: _pages[_selectedPageIndex]['page'],
+      bottomNavigationBar: BottomNavigationBar(
+        onTap: _selectPage,
+        backgroundColor: Theme.of(context).primaryColor,
```

Comment:
extract this to a variable

Reply:
extract it like this:

`ThemeData theme = Theme.of(context);`

then use it like this as an example on `backgroundColor` property,

`backgroundColor: theme.primaryColor`

Reply:
I am using ThemeData in the main.dart.
Kindly see the snip below.
![image](https://user-images.githubusercontent.com/74462760/102624009-5f91af80-417e-11eb-8023-8282cccf6ad7.png)

then, with the help of  'context', I am using the ThemeData from the main.dart to other widgets.
With these, should I still need to extract the ThemeData as you said?

Reply:
extracting it to a variable means you just have to declare it based on the given example. doing it that way avoids DRY principle (Don't Repeat Yourself), please take a look at these articles:

https://thevaluable.dev/dry-principle-cost-benefit-example/
https://medium.com/@nrk25693/dry-or-wet-and-why-867ac3096483

Reply:
the snip from main.dart can be used throughout the application, what you have done there is just setting preferential values which can be used by the routes/pages you declared.